libs/server/Resp/Vector/VectorManager.cs

```diff
@@ -0,0 +1,743 @@
+                    element,
+                    valueType,
+                    values,
+                    attributes
+                );
+
+            if (insert)
+            {
+                return VectorManagerResult.OK;
+            }
+
+            return VectorManagerResult.Duplicate;
+        }
+
+        /// <summary>
+        /// Try to remove a vector (and associated attributes) from a Vector Set, as identified by element key.
+        /// </summary>
+        internal VectorManagerResult TryRemove(ReadOnlySpan<byte> indexValue, ReadOnlySpan<byte> element)
+        {
+            AssertHaveStorageSession();
+
+            ReadIndex(indexValue, out var context, out _, out _, out var quantType, out _, out _, out var indexPtr, out _);
+
+            if (quantType == VectorQuantType.XPreQ8 && element.Length != sizeof(int))
+            {
+                // We know this element isn't present because of other validation constraints, bail
+                return VectorManagerResult.MissingElement;
+            }
+
+            var del = Service.Remove(context, indexPtr, element);
+
+            return del ? VectorManagerResult.OK : VectorManagerResult.MissingElement;
+        }
+
+        /// <summary>
+        /// Deletion of a Vector Set needs special handling.
+        /// 
+        /// This is called by DEL and UNLINK after a naive delete fails for us to _try_ and delete a Vector Set.
+        /// </summary>
+        internal Status TryDeleteVectorSet(StorageSession storageSession, ref SpanByte key)
+        {
+            storageSession.parseState.InitializeWithArgument(ArgSlice.FromPinnedSpan(key.AsReadOnlySpan()));
+
+            var input = new RawStringInput(RespCommand.VADD, ref storageSession.parseState);
+
+            Span<byte> indexSpan = stackalloc byte[IndexSizeBytes];
+
+            using (ReadForDeleteVectorIndex(storageSession, ref key, ref input, indexSpan, out var status))
+            {
+                if (status != GarnetStatus.OK)
+                {
+                    // This can happen is something else successfully deleted before we acquired the lock
+                    return Status.CreateNotFound();
+                }
+
+                DropIndex(indexSpan);
+
+                // Update the index to be delete-able
+                var updateToDroppableVectorSet = new RawStringInput();
+                updateToDroppableVectorSet.arg1 = DeleteAfterDropArg;
+                updateToDroppableVectorSet.header.cmd = RespCommand.VADD;
+
+                var update = storageSession.basicContext.RMW(ref key, ref updateToDroppableVectorSet);
+                if (!update.IsCompletedSuccessfully)
+                {
+                    throw new GarnetException("Failed to make Vector Set delete-able, this should never happen but will leave vector sets corrupted");
+                }
+
+                // Actually delete the value
+                var del = storageSession.basicContext.Delete(ref key);
```

Comment:
For my learning: what happens if a crash or something happens in the middle of  `TryDeleteVectorSet` - That VectorSet will be corrupted? Or from the AOF, Garnet will be able to recover to a valid state?